plugins/modules/openssh_keypair.py

```diff
@@ -180,6 +212,57 @@ def __init__(self, module):
         if self.regenerate == 'always':
             self.force = True
 
+        # The empty string is intentionally ignored so that dependency checks do not cause unnecessary failure
+        if self.passphrase:
+            if not HAS_OPENSSH_SUPPORT:
+                module.fail_json(
+                    msg=missing_required_lib(
+                        'cryptography >= 2.6',
+                        reason="to encrypt/decrypt private keys with passphrases"
+                    )
+                )
+
+            ssh = module.get_bin_path('ssh', True)
+            proc = module.run_command([ssh, '-Vq'])
+            ssh_version = parse_openssh_version(proc[2].strip())
+
+            if module.params['private_key_format'] == 'auto':
+                if LooseVersion(ssh_version) >= LooseVersion("7.8") and not HAS_OPENSSH_PRIVATE_FORMAT:
+                    module.fail_json(
+                        msg=missing_required_lib(
+                            'cryptography >= 3.0',
+                            reason="to load/dump private keys in the default OpenSSH format for OpenSSH >= 7.8"
+                        )
+                    )
+                elif LooseVersion(ssh_version) < LooseVersion("7.8"):
+                    if self.type == 'ed25519':
+                        if HAS_OPENSSH_PRIVATE_FORMAT:
+                            # ed25519 keys are always formatted in the OpenSSH format by ssh-keygen
+                            # since their introduction in OpenSSH 6.5
+                            self.private_key_format = 'SSH'
+                        else:
+                            module.fail_json(
+                                msg=missing_required_lib(
+                                    'cryptography >= 3.0',
+                                    reason="to load/dump ed25519 private keys in the default OpenSSH format"
+                                )
+                            )
+                    else:
+                        # OpenSSH made SSH formatted private keys available in version 6.5,
+                        # but still defaulted to PKCS1 format
+                        self.private_key_format = 'PKCS1'
+                else:
+                    self.private_key_format = 'SSH'
```

Comment:
Isn't this basically equivalent to:
```suggestion
                self.private_key_format = 'SSH'
                if LooseVersion(ssh_version) < LooseVersion("7.8") and self.type != 'ed25519':
                    # OpenSSH made SSH formatted private keys available in version 6.5,
                    # but still defaulted to PKCS1 format
                    self.private_key_format = 'PKCS1'

                if self.private_key_format == 'SSH' and not HAS_OPENSSH_PRIVATE_FORMAT:
                    module.fail_json(
                        msg=missing_required_lib(
                            'cryptography >= 3.0',
                            reason="to load/dump private keys in the default OpenSSH format for OpenSSH >= 7.8 or for ed25519 keys"
                        )
                    )
```
?